### PR TITLE
TPC: add mergeable output for Clusters task

### PR DESCRIPTION
**Pull request by tklemenz**
The Clusters task now has the option to use mergeable output to be able to run the task locally and do only the merging on the remote node. For the visualization of the merged output the post-processing task `ClusterVisualizer` is needed.

The Clusters task can still be run in the exact same way before if needed.

An exemplary json file for mergeable output in a local Clusters task is `Modules/TPC/run/tpcQCClusters_direct.json`. The `taskParameter` `mergeableOutput` has to be set to true (false is default even if the option is completely omitted).

### CODE REVIEW

**Review by knopers8 — COMMENTED** [review]
Just to be sure you are aware, there is a way to let QCG visualize a custom class which inherits from a ROOT class:
https://github.com/AliceO2Group/QualityControl/blob/master/doc/Advanced.md#display-a-non-standard-root-object-in-qcg
Perhaps this way you could avoid using a postprocessing task to display the results, though i did not follow your code in detail

**Comment by tklemenz** [comment]
Hello Piotr, thanks for the hint but I think in this case it is not possible to do it like this. In our case we have a data object `ClustersData` (mergeable) containing multiple `CalDet` objects where in the visualizer for each of them three canvases (containing multiple histograms) are created and displayed in the QCG. For example like `qc/TPC/MO/Clusters/c_Sides_Q_Max` in the test QCG. I think there is no other way to do it.

File: Modules/TPC/include/TPC/ClustersData.h [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+namespace o2::quality_control_modules::tpc
+{
+
+class ClustersData final : public TObject, public MergeInterface
+{
+ public:
+  ClustersData()
+    : o2::mergers::MergeInterface(), TObject()
+  {
+  }
```
Comment (wiechula) [inline Modules/TPC/include/TPC/ClustersData.h:41]: Perhaps the constructor should take the name as argument, which defaults to `ClusterData`. If we want to use it for the Digits as well it should have a different name.

File: Modules/TPC/src/ClusterVisualizer.cxx [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+      }
+    }
+  }
+
+  mPath = config.get<std::string>("qc.postprocessing." + name + ".path");
+}
+
+void ClusterVisualizer::initialize(Trigger, framework::ServiceRegistry&)
+{
+  mCdbApi.init("ccdb-test.cern.ch:8080");
```
Comment (wiechula) [inline Modules/TPC/src/ClusterVisualizer.cxx:115]: Shouldn't this be configurable from the json?

**Review by wiechula — CHANGES_REQUESTED** [review]
Just small things.

File: Modules/TPC/include/TPC/ClustersData.h (hunk 1 above)
Comment (tklemenz) [reply Modules/TPC/include/TPC/ClustersData.h:41]: I was actually thinking to do a separate class for the RawDigits task but I can try to do it within this one.

File: Modules/TPC/src/ClusterVisualizer.cxx (hunk 2 above)
Comment (tklemenz) [reply Modules/TPC/src/ClusterVisualizer.cxx:115]: absolutely

File: Modules/TPC/include/TPC/ClustersData.h (hunk 1 above)
Comment (tklemenz) [reply Modules/TPC/include/TPC/ClustersData.h:41]: I implemented a setName function that is called in the initialize function of the Clusters/RawDigits task to set the proper name of the data object.

**Comment by tklemenz** [comment]
This PR needs https://github.com/AliceO2Group/AliceO2/pull/7792 from O2 now.

File: Modules/TPC/include/TPC/Clusters.h [hunk 3]
```diff
@@ -54,7 +54,8 @@ class Clusters /*final*/ : public TaskInterface // todo add back the "final" whe
   void reset() override;
 
  private:
-  o2::tpc::qc::Clusters mQCClusters{};                         ///< O2 Cluster task to perform actions on cluster objects
+  bool mIsMergeable = true;
+  ClustersData mQCClusters{ "N_Clusters" };                    ///< O2 Cluster task to perform actions on cluster objects
```
Comment (wiechula) [inline Modules/TPC/include/TPC/Clusters.h:58]:
```suggestion
  ClustersData mQCClusters{};                    ///< O2 Cluster task to perform actions on cluster objects
```
The default should be "N_Clusters" so this is not really needed.

**Review by wiechula — APPROVED** [review]
Just one very minor thing, which could also be fixed later. Please remove the [WIP]

File: Modules/TPC/include/TPC/Clusters.h (hunk 3 above)
Comment (wiechula) [reply Modules/TPC/include/TPC/Clusters.h:58]: But it also doesn't hurt to leave it like this...

Comment (tklemenz) [reply Modules/TPC/include/TPC/Clusters.h]: done

**Review by wiechula — APPROVED** [review]

**Comment by wiechula** [comment]
@knopers8 , I think the failure in the CI is unrelated. Could you have a look?

**Pull request merged**